server/store/sqlstore/legal_hold.go

```diff
@@ -110,14 +110,24 @@ func (ss SQLStore) GetPostsBatch(channelID string, endTime int64, cursor model.L
 // GetChannelIDsForUserDuring gets the channel IDs for all channels that the user indicated by userID is
 // a member of during the time period from (and including) the startTime up until (but not including) the
 // endTime.
-func (ss SQLStore) GetChannelIDsForUserDuring(userID string, startTime int64, endTime int64) ([]string, error) {
+func (ss SQLStore) GetChannelIDsForUserDuring(userID string, startTime int64, endTime int64, excludePublic bool) ([]string, error) {
 	query := ss.replicaBuilder.
 		Select("distinct(cmh.channelid)").
-		From("channelmemberhistory as cmh").
+		From("channelmemberhistory as cmh")
+
+	query = query.
```

Comment:
What's the improvement of this change?

Reply:
That's just me making some tests. PR is in draft thus not ready to review. Will attach screenshots once finished, when the UI is somewhat pretty.